examples/result_file_dataclasses.py

```diff
@@ -0,0 +1,202 @@
+"""
+Working with result files using Python-native dataclasses.
+"""
+from collections.abc import Iterator
+from operator import attrgetter
+from pathlib import Path
+
+import pendulum
+import polars as pl
+
+from indico import IndicoClient
+from indico.queries import (
+    GetSubmission,
+    SubmissionResult,
+    RetrieveStorageObject,
+    SubmitReview,
+)
+from indico_toolkit import results
+
+
+"""
+Load a result file.
+"""
+client = IndicoClient()
+submission = client.call(GetSubmission(123))  # Or similar.
+result_dict = client.call(RetrieveStorageObject(submission.result_file))
+result = results.load(result_dict)
+
+# If your submission is unreviewed,
+# generate a result file to include review information.
+submission_result = client.call(SubmissionResult(submission, wait=True))
+result_dict = client.call(RetrieveStorageObject(submission_result.result))
+result = results.load(result_dict)
+
+# Or convert the unreviewed submission to a reviewed submission.
+# Due to the lack of review data, only `document.final` is populated.
+result = results.load(result_dict, convert_unreviewed=True)
+
+# `results.load` also takes JSON strings and file paths.
+for result_file in Path("results_folder").glob("*.json"):
+    result = results.load(result_file)
+
+
+"""
+Save document classification and extractions to CSV.
+"""
+
+
+def get_predictions(result: results.Result) -> Iterator[dict[str, object]]:
+    yield {
+        "submission_id": result.submission_id,
+        "field": "Classification",
+        "value": result.document.final.classification.label,
+        "confidence": result.document.final.classification.confidence,
+    }
+
+    for extraction in result.document.final.extractions:
+        yield {
+            "submission_id": result.submission_id,
+            "field": extraction.label,
+            "value": extraction.text,
+            "confidence": extraction.confidence,
+        }
+
+
+pl.DataFrame(get_predictions(result)).write_csv("predictions.csv")
```

Comment:
curious what your thoughts are on putting this functionality into the class itself?

Reply:
I'm hesitant to add more functionality to the dataclasses, as they're already quite large. My guiding principle was to create the kind of result API that I'd like to use to write higher level code like this.

I've been thinking that it might be better to group that kind of functionality in other modules. For example, a downselection function that uses `groupby()`, `orderby()` and `reject()` underneath would makes sense to have in the auto review module. And that keeps these classes from ending up with all manner of auto review, post processing, and prediction dump code in them.

Of course I'm still very open to adding new functions to the classes if they'd be generally useful to all those cases--things we find ourselves writing repeatedly and would be nice to have generalized as a part of the result API itself.

Reply:
makes sense!